### DISCUSSION TITLE
Inline snaphots do not update when ran in vs code or terminal with swift tests when using new swift-testing

### DISCUSSION THREAD

**Discussion by doozMen** [pointfreeco/swift-snapshot-testing#930] [General]
**Describe the bug**
See reproducer project https://github.com/doozMen/swift-snapshot-xcodebuild 

**To Reproduce**

The reproducer project has description in readme.md. But in short

1. make an inline snaphot test using swift testing
2. set recording to `.all` or parameter `record: true`
3. run `swift test`
   > note: nothing happens
5. run `xcodebuild -scheme <#package#> test -destination "platform=macOS" -quiet
  > note: test is updated

**Expected behavior**

Both inline and file tests should update, but they do not

**Environment**
  - swift-snapshot-testing version 1.17.6
  - Xcode 16.1
  - Swift 6.0.2
  - OS: macOS 15

**Comment by mbrandonw** [comment]
Hi @doozMen, the snapshot recording mechanism depends on `XCTestObservationCenter` of which Swift Testing has no version right now. They are aware of the [issue](https://github.com/swiftlang/swift-testing/issues/469), but without this there isn't much we can do. Recording for inline snapshots will just need to be done through Xcode for the time being.

Since this isn't an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there!

**Reply by mbrandonw** [reply]
We could provide custom logic for Swift Testing and write out inline snapshots after _every_ test run instead of all at once at the end of the test suite. But then that would also force tests to be run in serial because by default Swift Testing tests run in parallel and that could make it so that multiple recording tests are writing to the same file at once.

**Reply by mbrandonw** [reply]
Sorry, after playing with this a bit I don't think it's possible to actually write inline snapshots one at a time. It really does seem to need to be done all at once at the end of the test run, and so we are stuck waiting for Swift Testing support.

**Reply by Almaz5200** [reply]
It's a shame their fix is still not even close to being released :(
